We fixed this for 0.77 for non-shadowed cases, but I missed this one.
--- Linux-PAM-0.77/modules/pam_unix/pam_unix_passwd.c	2004-02-12 15:39:27.000000000 -0500
+++ Linux-PAM-0.77/modules/pam_unix/pam_unix_passwd.c	2004-02-12 15:41:56.000000000 -0500
@@ -439,7 +439,7 @@
     }
 }
 
-static int _update_shadow(const char *forwho, char *towhat)
+static int _update_shadow(pam_handle_t *pamh, const char *forwho, char *towhat)
 {
     struct spwd *spwdent = NULL, *stmpent = NULL;
     FILE *pwfile, *opwfile;
@@ -508,6 +508,7 @@
 
     if (!err) {
 	rename(SH_TMPFILE, "/etc/shadow");
+	_log_err(LOG_NOTICE, pamh, "password changed for %s", forwho);
 	return PAM_SUCCESS;
     } else {
 	unlink(SH_TMPFILE);
@@ -598,7 +599,7 @@
 	}
 	if (_unix_comesfromsource(pamh, forwho, 1, 0)) {
 		if (on(UNIX_SHADOW, ctrl) || _unix_shadowed(pwd)) {
-			retval = _update_shadow(forwho, towhat);
+			retval = _update_shadow(pamh, forwho, towhat);
 			if (retval == PAM_SUCCESS)
 				if (!_unix_shadowed(pwd))
 					retval = _update_passwd(pamh, forwho, "x");
